tests/e2e/framework/kubernetes.go

```diff
@@ -142,6 +154,48 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
 	} else {
 		releaseDir = util.GetResourcePath("")
 	}
+	var kubeConfig, remoteKubeConfig string
+	var kubeClient, remoteKubeClient kubernetes.Interface
+	if *multiClusterDir != "" {
+		// ClusterRegistiresDir indicates the Kubernetes cluster config should come from files versus KUBECONFIG
+		// environmental variable.  The test config can be defined to use either a single cluster or 2 clusters
+		var clusterStore *clusterregistry.ClusterStore
+		clusterStore, err = clusterregistry.ReadClusters(*multiClusterDir)
+		if clusterStore == nil {
+			log.Errorf("Failed to clusters in the ClusterRegistriesDir %s\n", *multiClusterDir)
+			return nil, err
+		}
+		if clusterStore != nil {
+			kubeConfig = clusterStore.GetPilotAccessConfig()
+			kubeConfig = path.Join(*multiClusterDir, kubeConfig)
+			//				kubeConfig = kubeCfgFile
+			if _, kubeClient, err = kube.CreateInterface(kubeConfig); err != nil {
+				return nil, err
+			}
+			// Note only a single remote cluster is currently supported.
+			clusters := clusterStore.GetPilotClusters()
+			for _, cluster := range clusters {
+				kubeconfig := clusterregistry.GetClusterAccessConfig(cluster)
+				remoteKubeConfig = path.Join(*multiClusterDir, kubeconfig)
+				log.Infof("Cluster name: %s, AccessConfigFile: %s", clusterregistry.GetClusterName(cluster), remoteKubeConfig)
+				// Expecting only a single remote cluster so hard code this.  The code won't throw an error
+				// if more than 2 clusters are defined in the config files, but will only use the last cluster parsed.
+				if _, remoteKubeClient, err = kube.CreateInterface(remoteKubeConfig); err != nil {
+					return nil, err
+				}
+			}
+		}
+	} else {
+		tmpfile := *namespace + "_kubeconfig"
+		tmpfile = path.Join(tmpDir, tmpfile)
+		if err = util.GetKubeConfig(tmpfile); err != nil {
+			return nil, err
+		}
+		kubeConfig = tmpfile
+	}
+
+	a := NewAppManager(tmpDir, *namespace, i, kubeConfig)
+
```

Comment:
I guess that an appManager for the remote cluster is also needed as part of this patch?